packages/evm/lib/nft/EvmNftProvider.ts

```diff
@@ -135,15 +93,25 @@ export abstract class EvmNftProvider extends Nft<BaseProvider, Signer> {
         return await contract.isApprovedForAll(owner.toString(), operator.toString());
     }
 
-    public async approveAll(
+    public async estimateTransfer(
         contractAddress: AddressType,
-        operator: AddressType,
-        state: boolean,
-        fee?: FeeType
-    ): Promise<Transaction<EthersTransactionResponse>> {
-        const { contract } = await this._cacheGet(contractAddress);
-        const tx = await contract.populateTransaction.setApprovalForAll(operator.toString(), state);
-        return this.walletProvider.sendTransaction(toEthereumTxRequest(tx, fee));
+        receiver: AddressType,
+        tokenIDs: string[],
+        amounts?: number[],
+        data = '0x'
+    ): Promise<BigNumber> {
+        const tx = await this.populateTrasnfer(contractAddress, receiver, tokenIDs, amounts, data);
+        return new BigNumber((await (this.walletProvider as EvmBaseWalletProvider<BaseProvider>).estimateGas(tx)).toString());
+    }
+
+    public async estimateApprove(contractAddress: AddressType, operator: AddressType, tokenID: number): Promise<BigNumber> {
+        const tx = await this.populateApprove(contractAddress, operator, tokenID);
+        return new BigNumber((this.walletProvider as EvmBaseWalletProvider<BaseProvider>).estimateGas(tx).toString());
```

Comment:
break this into multiple lines for better readability, also why the casting to EvmBaseWalletProvider ?